dashboard/views.py

```diff
@@ -134,6 +137,90 @@ def get_course_info(request, course_id=0):
     return HttpResponse(json.dumps(resp, default=str))
 
 
+@permission_required('dashboard.update_course_info',
+                     fn=objectgetter(Course, 'course_id', 'canvas_id'), raise_exception=True)
+def update_course_info(request, course_id=0):
+    """
+
+    :param request: HTTP `PUT` req.; body should contain the JSON body…
+    :param course_id: Integer Canvas course ID number, typically six digits or less.
+    :return: HttpResponse containing `{"default": "success"}` or `{"default": "fail"}`
+    """
+    logger.info(update_course_info.__name__)
+
+    if (request.method != 'PUT'):
+        return HttpResponseBadRequest(JsonResponse({'error': 'Invalid request method.'}))
+
+    course_id = canvas_id_to_incremented_id(course_id)
+    current_user = request.user.get_username()
+
+    bad_json_response = HttpResponseBadRequest(JsonResponse({'error': 'Request JSON malformed.'}))
+
+    try:
+        request_data: dict = json.loads(request.body.decode('utf-8'))
+    except JSONDecodeError:
+        return bad_json_response
+
+    schema = {'$schema': 'http://json-schema.org/draft-07/schema',
```

Comment:
Yes, I had given that a little thought as I was writing it.

I think that the schema is so specific to this API entry that it should stay here.  If the schema could be reduced in size (but not readability) and still work the same way, I'd feel that even more strongly.

If the schema were to be used by multiple API entries it would make sense to move it to a file.

So, I think I'll leave it here.  If you have any suggestions for improving its formatting or reducing it, please tell me, I'd love to know.

Reply:
I'll leave it up to you. `views.py` is already gigantic, so anything to keep things compact is on my radar. I have formatting preferences in general, but they would make it longer, not shorter.